scripts/git/update_to_llvm_syncpoint.py

```diff
@@ -101,12 +116,14 @@ def parse_arguments():
     args.repo = utils.find_git_toplevel()
 
   # Set some defaults.
-  if not args.tensorflow_path:
-    args.tensorflow_path = os.path.join(args.repo, "third_party", "tensorflow")
   if not args.llvm_path:
     args.llvm_path = os.path.join(args.repo, "third_party", "llvm-project")
   if not args.llvm_bazel_path:
     args.llvm_bazel_path = os.path.join(args.repo, "third_party", "llvm-bazel")
+  if not args.tensorflow_path:
+    args.tensorflow_path = os.path.join(args.repo, "third_party", "tensorflow")
+  if not args.mlir_hlo_path:
+    args.mlir_hlo_path = os.path.join(args.repo, "third_party", "mlir-hlo")
 
   return args
 
```

Comment:
(Pretend this comment is on `def main(args):` two lines below...)
AFAICT there isn't anything in google's python style or pep8 that says that the `main` function should be at the end of the file (or anywhere else), but it's done so universally that it feels strange to have the `main` function here. I guess there could be an argument for having it close to the argument parsing, but it's also odd to see functions called before they're defined.

Not asking for a change, just find it strange that's not mentioned anywhere in the style guides.

Reply:
Yeah I think this is the classic question of whether to do things top-down or bottom-up. If the functions have reasonable names, it should be possible to read main and understand basically what's going on without having read the definitions yet. IDK